notes/D00-collect_taxonomic_cogscore_data.jl

```diff
@@ -0,0 +1,101 @@
+#####
+# Notebook D00: Data collection and preparation notebook for analysis scripts
+# NOTE: this notebook will become functions on src/prediction.jl source in the future.
+#####
+
+## Loading packages
+
+using Resonance
+using CSV
+using Arrow
+using Dates
+using DataFrames
+
+## Collecting the data from local files
+resonance_datapath = get(ENV, "EXPORTED_DATA_PATH", "data")
+
+### 1. Sample and subject metadata
+
+timepoints_df = CSV.read(
+    joinpath(resonance_datapath, "timepoint_metadata.csv"),
+    DataFrame;
+    delim = ',',
+    types = [
+        Int64,      #subject
+        Int64,      #timepoint
+        Float64,    #ageMonths
+        String,     #race
+        String,     #maternalEd
+        Dates.Date, #assessmentDate
+        Dates.Date, #scanDate
+        Float64,    #cogScore
+        String,     #omni
+        String,     #etoh
+        Bool,       #has_segmentation
+        Float64,    #read_depth
+        ]
+)
+
+#### Assessing timepoints sanity
+# if any(.!ismissing.(timepoints_df.cogScore) .⊻ .!ismissing.(timepoints_df.assessmentDate))
+#     xor_idx = findall(.!ismissing.(timepoints_df.cogScore) .⊻ .!ismissing.(timepoints_df.assessmentDate))
+#     xor_filtered_df = timepoints_df[xor_idx, :]
+#     @warn "There are $(nrow(xor_filtered_df)) timepoint samples with an assessmentDate incompatible with cogScore:\n $xor_filtered_df"
+# end
+
+### 2. Taxonomic profiles
+
+taxonomic_data = DataFrame(Arrow.Table(joinpath(resonance_datapath, "taxa.arrow")))
```

Comment:
I saw your comment. I suggest we merge this current state (because `load_data.jl` was not available on Main when I created the branch) and then fix this notebook on the next PR from a fresh branch with the latest functions

Reply:
We can do this for now. In future, you can always merge from main to get to the most recent state. But let's deal with that another time.